docker-compose.yml

```diff
@@ -50,14 +51,96 @@ services:
       - ${BASE_PATH}:/zeitgeist/data
     command: --base-path=/zeitgeist/data --dev --ws-external --rpc-external --rpc-cors all --rpc-methods Unsafe
 
+  indexer-db:
+    container_name: indexer-db
+    image: cockroachdb/cockroach:v22.2.2
+    ports:
+      - $INDEXER_DB_PORT:26257
+    networks:
+      - subsquid
+    volumes:
+      - $BASE_PATH:/cockroach/cockroach-data
+    command: start-single-node --insecure
+
+  indexer:
+    container_name: indexer
+    image: subsquid/substrate-ingest:1
+    restart: unless-stopped
+    depends_on:
+      - zeitgeist-blockchain
+      - indexer-db
+    networks:
+      - blockchain
+      - subsquid
+    ports:
+      - $INDEXER_PROM_PORT:9090
+    command: -e ws://zeitgeist-node:$ZTG_CHAIN_WS_PORT -c 5 --out postgres://root@indexer-db:$INDEXER_DB_PORT/defaultdb
+
+  indexer-api:
+    container_name: indexer-api
+    image: subsquid/substrate-gateway:2.5.0
+    depends_on:
+      - indexer
+    networks:
+      - subsquid
+    environment:
+      RUST_LOG: "actix_web=info,actix_server=info"
+    ports:
+      - $INDEXER_API_PORT:8000
+    command: --database-url postgres://root@indexer-db:$INDEXER_DB_PORT/defaultdb --database-max-connections 10
+
+  processor-db:
+    container_name: processor-db
+    image: postgres:14
+    networks:
+      - subsquid
+    environment:
+      POSTGRES_DB: postgres
+      POSTGRES_PASSWORD: postgres
+    ports:
+      - $PROCESSOR_DB_PORT:5432
+
+  processor:
+    container_name: processor
+    image: zeitgeistpm/squid-processor:latest
+    depends_on:
+      - indexer-api
+      - processor-db
+      - ipfs
```

Comment:
It seems to depend on ipfs, but does not join the network of ipfs and does not configure ipfs api endpoint address anywhere. Given that configuration, it should be unable to use ipfs unless there is some autodiscovery mechanism in place.

Reply:
Shouldn't it also depend on the blockchain?

Reply:
From the logs I understand that one or more containers haven't been removed with `docker-compose down` or maybe your port `5432` is blocked by some other service.

About the dependency graph, processor -> indexer-api -> indexer -> zeitgeist-blockchain. So `processor` is indirectly dependent on `zeitgeist-blockchain`.